Parsers/ASimWebSession/Parsers/ASIMWebSessionIIS.yaml

```diff
@@ -0,0 +1,121 @@
+                Dst = sIP,
+                Src = cIP,
+                UserAgent = csUserAgent,
+                ThreatCategory = IndicatorThreatType,
+                EventOriginalSeverity = Severity,
+                SrcGeoCountry = RemoteIPCountry,
+                SrcGeoLatitude = RemoteIPLatitude,
+                SrcGeoLongitude = RemoteIPLongitude,
+                ThreatOriginalConfidence = Confidence,
+                ThreatIpAddr = MaliciousIP,
+                ThreatIsActive = IsActive,
+                ThreatFirstReportedTime = FirstReportedDateTime,
+                ThreatLastReportedTime = LastReportedDateTime,
+                EventReportUrl = ReportReferenceLink
+        | extend
+                SrcUsername = User,
+                DstNatIpAddr = iff(csHost <> "", Dst, ""),
+                EventType = 'WebServerSession',            
+                EventVendor = 'Microsoft',
+                EventSchemaVersion = '0.2.5',
+                EventSchema = 'WebSession', 
+                EventProduct = 'IIS',
+                DvcOs = 'Windows',
+                EventCount = int(1),
+                SrcIpAddr = Src,
+                IpAddr = Src,
+                HttpUserAgent = UserAgent,
+                HttpStatusCode = tostring(EventResultDetails),
+                EventStartTime = ( (TimeGenerated) - (TimeTaken * 1ms)), // TimeTaken field is in Milliseconds 
+                EventEndTime =  TimeGenerated,
+                EventSeverity = iff(EventResult == "Success", "Low", "Informational"),
+                csUriQuery = tostring(csUriQuery),
+                csUriStem = tostring(csUriStem),
+                tempURLconstruct = iff(csUriQuery == "", csUriStem, strcat(csUriStem,"?",csUriQuery)),
+                sPort = tostring(sPort),
+                HttpHost = iff ( HttpHost == "-", "", HttpHost),
+                csHost = iff ( csHost == "-", "", csHost),  //remove empty values
+                EventOriginalResultDetails = iff(scSubStatus <> "0", strcat (scStatus, ".", scSubStatus), scStatus)
+        | extend 
+                ipv6_parts = extract_all (@'^\[(.+)\](?:\:(\d+))?$',csHost)[0],
+                ipv4_parts = extract_all (@'^(\d+\.\d+\.\d+\.\d+)(?:\:(\d+))?$',csHost)[0],
+                host_parts = extract_all (@'^([^\\\d:]+)(?:\:(\d+))?$',csHost)[0]
+        | extend         
+                DstIpAddr = coalesce(ipv4_parts[0], ipv6_parts[0]),
+                DstPortNumber = coalesce(ipv4_parts[1], ipv6_parts[1], host_parts[1]),
+                HttpHost = coalesce (host_parts[0], HttpHost)
+        | project-away ipv4_parts, ipv6_parts, host_parts              
+        | extend 
+                schema = iff (sPort has "443", "https://", "http://"),
+                port = iff (sPort in ("80", "443"), "", strcat (":", sPort)),
+                host = case ( 
+                HttpHost == "Default Web Site", Dst,
+                HttpHost != "", HttpHost,
+                Dst
+               )
```

Comment:
Sounds you don't need 3 options (and hence a case). If HttPost is "", you use Dst.